### ISSUE TITLE
bundling ACs (from Richard)

### ISSUE THREAD

**Issue by boucadair** [boucadair/attachment-circuit-model#14]
More than the naming convention, we should also clarify the use case. There is - imho – a need for being able to grouping acs and being able to manipulate a set of ac (Think for example of a set of compute nodes attached to a TOR/Fabric or Cell Site Router). 
My thinking goes to something like “ac-bundle”, which can be a reference point for a connection to a service instead of the ac itself (think of a kind virtual interface). So this is more than a profile.

Example:

```
Comp1-------[  TOR  ]
Comp2---------|| |
Comp3----------| | 
...              |   
CompX------------| 

```
Assume you have sapX = CompX interface

We could get something like (spare me the approximative yaml, this is just for expressing the idea ).

```
ac-bundle:
name: aggregate-service-foo
l2-connection:
   cvlan-id: 100
ip-connection:
   ipv4:
     local-address: 192.168.1.1/24
     customer-addresses:
       address-pool:192.168.1.3-254

ac1
peer-sap sap1
ac-bundle-ref:
    aggregate-service-foo
ac2
peer-sap sap2
ac-bundle-ref:
    aggregate-service-foo
```

etc.

Basically, we’re getting an indirection between the service definition (rather static) and the connection (ac).
Next, with appropriate augmentation, we can anchor a service (L3SM, etc..) to a stable bundle identifier.  I am insisting on stable, since things move a lot in container ecosystem: you don’t really know in advance to which Node a service can be created, and you don’t know which address will be assigned. Here you may just add/remove your ac/sap to the bundle and you’re done.

**Comment by boucadair** [comment]
From @sbarguil:

> One of the doubts that I have from this approach, is how the Transport network controller should unwrap this global definition into network configurations. 
> 
> And the example you have provided is excellent. 
> 
> ac-bundle:
> name: aggregate-service-foo
> l2-connection:
>    cvlan-id: 100
> ip-connection:
>    ipv4:
>      local-address: 192.168.1.1/24
>      customer-addresses:
>        address-pool:192.168.1.3-254
>        assignation-policy: by-service
> 
> ac1
> peer-sap sap1
> ac-bundle-ref:
>     aggregate-service-foo
> ac2
> peer-sap sap2
> ac-bundle-ref:
>     aggregate-service-foo
> 
> My understanding is in this case:
> •	The controller needs to assign two interfaces from the network.
> o	How the controller determine what are the network interfaces it should reserve, in which node?
> 	The option I have seen from Med Example is the Bearer or the SAP. So this kind of requirements must always require a Bearer/SAP to be able to be applied on the network. Do you have any other idea about it?
> •	The controller needs to assign and reserve two IP addresses pool 192.168.1.3-254.
> o	How the controller determine what the free addresses of this pool? 
> o	As addresses can be reused across services do you think the requirement needs to have a service association? Or with bearer this can be inferred? In that case the bearer seems relevant too.
> o	In other case, for infra creation: Does the controller needs to read the whole network to determine which addresses are free in this pool.
> o	I understand If the size of the mask changed, for example is not /24 but /30 the controller should follow a different assignation rules. So maybe, we can consider add a leaf to determine IP address assignation pool behavior. 
> 	assignation-policy: by-service : it Means we expect this address is assigned in the context of the service attached.
> Regards,

**Comment by boucadair** [comment]
I think this is an sub case of ac-group-profile usage. It would be great to have an appendix that basically echoes the example in this issues and clarifies the various questions from Samier.

**Comment by boucadair** [comment]
An example will be added in #111
